block-chain/src/main/java/itx/blockchain/signed/SignedData.java

```diff
@@ -0,0 +1,26 @@
+package itx.blockchain.signed;
+
+/**
+ * Data object for storing signed data.
+ */
+public interface SignedData {
+
+    /**
+     * Algorithm name used for data signature.
+     * @return
+     */
+    String getAlgorithm();
+
+    /**
+     * Data payload. (any user data)
+     * @return
+     */
+    byte[] getData();
+
+    /**
+     * Digital signature of data payload.
+     * @return
+     */
+    byte[] getSignature();
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider adding validation annotations to interface</b></div>
<div id="fix">

Consider adding input validation annotations like `@NotNull` to the interface methods to enforce non-null return values for `getAlgorithm()`, `getData()` and `getSignature()`.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<blockquote>Check the AI-generated fix before applying</blockquote>
<div id="code">


```
 @@ -12,13 +12,16 @@
 -    String getAlgorithm();
 -    byte[] getData();
 -    byte[] getSignature();
 +    @NotNull
 +    String getAlgorithm();
 +    @NotNull
 +    byte[] getData();
 +    @NotNull
 +    byte[] getSignature();
```

</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/rkj6124/bito-test/pull/4#issuecomment-2557450846>#0dafab</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged